zonemaster/zonemaster-gui
# make navigation more consistent

## Purpose

Fix inconsistency issues on navigation.

## Context

Fixes #284

## Changes

* The domain checking part of the `domain` component is moved to a new `domain-check` component.
* The `domain` component is only used as a landing page and only display the `domain` component and the about section. No logic is implemented there.
* The `domain-check` component *actually* redirect to the `result` page (as opposed to only changing the url like it was doing before)
* The `result` component embed the `domain-check` component when it is accessed after a test finished
* The link in the navigation menu is no longer "active" when viewing result after being redirected from the form page

## How to test this PR

* Create a new test
* Verify that when the test finishes the browser is redirect to the result page with an url `/result/<id1>`. The domain check form is visible.
* Press the back button in the browser.
* Verify that only the domain check for is displayed. The url should be `/domain_check`.
* Press the forward button in the browser.
* Verify that the previous test result is displayed with the url `/result/<id1>`. The domain check form should still be visible.
* Create a second test for an other domain from the result page.
* When the second test finishes the url should change to `/result/<id2>`. The result for the second test should be displayed and the domain check form should still be visible.
* Refresh the page.
* The domain check form is not visible, instead a `Result` header is displayed.
